### PR TITLE
`CreateTask` process manager

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR introduces an example usage of the third major Spine entity - a `ProcessManager`.

The `CreateTaskWizard` supervises the flow of creating a task with all its fields.
For that, the procman handles a set of commands, which primarily cause the procman to post other commands. For example, a single `AddLabels` command may cause the procman to post `CreateBasicLabel`, `UpdateLabel` and `AssignLabel` commands.

The task creation process relies on a state machine to validate the received commands.
The state map of a `TaskCreation` process looks like:
```
--> Task Definition --> Label Assignment -->  Confirmation -->  Completed.
                 \                    \                 \
                  -------------------------------------------> Canceled.
```
**`Task Definition`** stage is where the primary task data (like the description and the due date) is assigned to the task.
**`Label Assignment`** stage is where the labels are created and assigned to the task. This stage must go after **`Task Definition`**. 
**`Confirmation`** stage is the stage where the user checks if the data is correct.
**`Completed`** and **`Canceled`** stages are the terminal stages of a process. When a process moves to a terminal stage, the `CreateTaskWizard` procman is being archived.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Comment by armiol** [comment]
`Task Definition` is better for the first step. If that can be confused with something else, so be it.

`Task Building` is the title for the whole process, not for its first part.

File: api-java/src/main/java/io/spine/examples/todolist/c/procman/TaskCreationWizard.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+ *     {@code
+ *     --> Task Building --> Label Assignment -->  Confirmation -->  Completed.
+ *                     \                    \                 \
+ *                      --------------------------------------------> Canceled.
+ *     }
+ * </pre>
+ *
+ * @author Dmytro Dashenkov
+ */
+@SuppressWarnings("unused") // Reflective access.
```
Comment (armiol) [inline api-java/src/main/java/io/spine/examples/todolist/c/procman/TaskCreationWizard.java]: Reflective access of what? Please be thorough. That's an example. 

`Message handlers are invoked by the framework via reflection.`

File: api-java/src/main/java/io/spine/examples/todolist/c/procman/WizardCommands.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            final TodoCommand updateDueDate = updateTaskDueDate(dueDate);
+            commands.add(updateDueDate);
+        }
+        return commands.build();
+    }
+
+    /**
+     * Creates a command updating the description of the target task to the given value.
+     *
+     * <p>It is implied that the task description is currently empty.
```
Comment (armiol) [inline api-java/src/main/java/io/spine/examples/todolist/c/procman/WizardCommands.java]: That's strange. Why would you mention that at all?

Same question applies to the rest of similar cases.

**Comment by armiol** [comment]
@dmdashenkov please make sure to NOT submit `Task Building`.

**Pull request merged**